model/SyncLog/Storage/RdsSyncLogStorage.php

```diff
@@ -0,0 +1,253 @@
+<?php
+ /**
+  * This program is free software; you can redistribute it and/or
+  * modify it under the terms of the GNU General Public License
+  * as published by the Free Software Foundation; under version 2
+  * of the License (non-upgradable).
+  *
+  * This program is distributed in the hope that it will be useful,
+  * but WITHOUT ANY WARRANTY; without even the implied warranty of
+  * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  * GNU General Public License for more details.
+  *
+  * You should have received a copy of the GNU General Public License
+  * along with this program; if not, write to the Free Software
+  * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+  *
+  * Copyright (c) 2018 (original work) Open Assessment Technologies SA ;
+  */
+
+ namespace oat\taoSync\model\SyncLog\Storage;
+
+ use Doctrine\DBAL\Query\QueryBuilder;
+ use oat\oatbox\service\exception\InvalidServiceManagerException;
+ use oat\oatbox\extension\script\MissingOptionException;
+ use oat\oatbox\service\ConfigurableService;
+ use oat\taoSync\model\SyncLog\SyncLogEntity;
+ use oat\taoSync\model\SyncLog\SyncLogFilter;
+ use common_exception_NotFound;
+ use common_persistence_SqlPersistence;
+ use Doctrine\DBAL\Connection;
+
+ class RdsSyncLogStorage extends ConfigurableService implements SyncLogStorageInterface
+ {
+     const SERVICE_ID = 'taoSync/RdsSyncLogStorage';
+
+     const OPTION_PERSISTENCE_ID = 'persistenceId';
+
+     const TABLE_NAME = 'synchronisation_log';
+
+     /**
+      * @var string
+      */
+     private $persistenceId = null;
+
+     /**
+      * @var common_persistence_SqlPersistence
+      */
+     private $persistence = null;
+
+     /**
+      * SyncLogService constructor.
+      * @param array $options
+      * @throws MissingOptionException
+      */
+     public function __construct(array $options = array())
```

Comment:
Is there a reason to give a default value to $options if at least one option (persistenceId) is mandatory for the class to be instantiated ?

Reply:
You are right, it doesn't make sense here. Fixed.